src/workers/nodeLayout.worker.ts

```diff
@@ -53,41 +53,16 @@ onmessage = async ({
       initialOverallTimeSpan,
     })
   }
-
-  if (graphData) {
-    const newData = JSON.parse(graphData) as GraphTimelineNode[]
-
-    if (timeScale && (layoutSetting || graphDataStore !== newData)) {
-      graphDataStore = prepareGraphData(newData)
-      await calculateNodeLayout()
-      const response: NodeLayoutWorkerResponseData = {
-        layout,
-        centerViewportAfter,
-      }
-      postMessage(response)
+  if (data && timeScale && currentLayoutSetting) {
```

Comment:
@pleek91 I updated this to check for `currentLayoutSetting` rather than a `layoutSetting` being passed in, since when data updates we don't pass the layout in again. Fixed the problem we had that was preventing us from merging this PR. I'ma merge it!!